### PR TITLE
readded append command

### PR DESCRIPTION
**Pull request by etp12**
## Overview

This command allows the user to append text to their previous message. The bot will resend the new message back to them. 

Example:
`User: http://google.`
`User: !!com`
`Bot: http://google.com`

If the new message is a command, the bot will also run it. This is useful if you forget a parameter of a command.

Example: 
`User: !gif`
`User: !! cats`
`Bot: !gif cats`

alt ex1

Test case still doesn't like the lines 
```
+import commands from '../index';
+import { callCmd } from '../../index';
```
In the source file. Removing these from /commands/useful/append.js will stop the error from happening when running `npm run mocha`

### CODE REVIEW

File: src/commands/useful/append.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  }
+
+  Promise.resolve(msg.channel.messages.getAll('author', msg.author))
+    .then(messages => {
+      messages.pop();
+      if (!messages[0]) {
+        return;
+      }
+
+      let newCmd = messages.pop().content.trim().concat(' ').concat(suffix);
```
Comment (dustinblackman) [inline src/commands/useful/append.js]: `const newCmd`

File: src/commands/useful/append.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      bot.sendMessage(msg.channel, newCmd);
+      parseMessage(bot, newCmd, msg);
```
Comment (dustinblackman) [inline src/commands/useful/append.js]: I'm confused. Why are you both sending the new command as a message and passing it to `parseMessage` to execute it?

File: src/commands/useful/append.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    })
+    .catch(err => {
+      sentry.captureError(err);
```
Comment (dustinblackman) [inline src/commands/useful/append.js]: It's just `sentry(err)` now.

File: src/commands/useful/append.js (hunk 2 above)
Comment (etp12) [inline src/commands/useful/append.js]: This way I was also allowing it to be used for regular text too, things that are not commands. This also allowed the bot to show the full command it is executing when using after appening something. Do you think it should just be `parseMessage()`?

Comment (dustinblackman) [inline src/commands/useful/append.js]: I would of thought this would of just been for commands, yes.

**Comment by dustinblackman** [comment]
Looks good, thanks! Going to do some local testing tomorrow and then merge. :)

**Comment by dustinblackman** [comment]
@etp12 Sorry for the delay. Just realized the tests are commented out. Mind fixing those up and I'll merge. :) All my local tests worked great!

**Comment by dustinblackman** [comment]
@etp12 Any update?

**Comment by etp12** [comment]
Still getting a strange error when trying to run the test I had that I am still working on. The imports for `commands` and `{callCmd}` seem to be the cause.

**Comment by dustinblackman** [comment]
@etp12 This with the latest master? I believe I had the same issue and added a workaround to the `npm run test` script.

**Comment by etp12** [comment]
@dustinblackman No it's not, I'll try in the next day or two when I go on break and get back to you.

**Comment by etp12** [comment]
@dustinblackman Still the same issue. The command itself works fine when run and passes lint fine, but running the test still fails [http://pastebin.com/hXzDxcAC](http://pastebin.com/hXzDxcAC). Seems like the test case can't import the command list from src/commands/index.js but I'm not sure.

**Comment by dustinblackman** [comment]
@etp12 I can experiment with the problem. Can you update your branch with master and push?

**Pull request closed without merging**